### PR TITLE
Make isKeyboardShortcut actually settable from the handler, and impro…

### PR DESCRIPTION
**Pull request by rassilon**
…ve the IKeyboardHandler documentation.

@amaitland , thoughts for 2272?

I was thinking any more would need to wait for 2357. :frowning: 

Bill

### CODE REVIEW

**Comment by amaitland** [comment]
> @amaitland , thoughts for 2272?

As in you'd like to merge this into `master`? I'm fine with that.

**Comment by rassilon** [comment]
I've added a sample that works to WinForms.Example. Not utterly complete, but good enough for most WinForm accelerator interactions I would think. It was easier than I thought. 

Bill

File: CefSharp/Internals/BrowserProcessServiceHost.cs [hunk 1]
```diff
@@ -52,13 +52,15 @@ public void SetOperationContext(OperationContext operationContext)
...
+            // Use TaskScheduler.Default to ensure that the work occurs
+            // on a background thread.
             return operationContextTask.ContinueWith(t =>
             {
                 var context = t.Result;
                 var renderProcess = context.GetCallbackChannel<IRenderProcess>();
                 var asyncResult = renderProcess.BeginEvaluateScriptAsync(browserId, frameId, script, timeout, null, null);
                 return Task.Factory.FromAsync<JavascriptResponse>(asyncResult, renderProcess.EndEvaluateScriptAsync);
-            }).Unwrap();
+            }, TaskScheduler.Default).Unwrap();
```
Comment (rassilon) [inline CefSharp/Internals/BrowserProcessServiceHost.cs]: Bah. I didn't mean to push this up. Seems pretty boring though. Thoughts? The idea is we're not guaranteed that the continuation (`ContinueWith`) *WILL NOT* execute on the current thread.

Comment (amaitland) [inline CefSharp/Internals/BrowserProcessServiceHost.cs]: Would have been ideal as a separate commit, oh well :smile: 

Makes sense to execute on the `ThreadPool`. I think adding something to the documentation would also be worthwhile.

File: CefSharp.Core/Internals/ClientAdapter.cpp [hunk 2]
```diff
@@ -195,8 +195,14 @@ namespace CefSharp
             {
                 return false;
             }
+            System::Boolean isKeyboardShortcut(*is_keyboard_shortcut);
```
Comment (amaitland) [inline CefSharp.Core/Internals/ClientAdapter.cpp]: I didn't think it was necessary to do anything more than pass in `*is_keyboard_shortcut` as a `ref` param?

At least it appeared to be working when I added `no_javascript_access` to `OnBeforePopup`

https://github.com/cefsharp/CefSharp/pull/935/files#diff-2731f23ba4e8fcf96f8501a49c6d687eR53

Comment (rassilon) [inline CefSharp.Core/Internals/ClientAdapter.cpp]: It isn't, updated.

**Comment by rassilon** [comment]
This fixes #1067 enough for me for the time being.

**Comment by rassilon** [comment]
Updated 41 WIP in Changelog.

Fyi,
Bill

**Comment by amaitland** [comment]
Nice :+1:

**Pull request merged**